### ISSUE TITLE
Add configuration for maintenance date/time window

### ISSUE THREAD

**Issue by camilonova** [TwinProduction/gatus#74]
Gatus is awesome. There is a way to setup a maintenance window to avoid being notified if a server (or many) goes down?

**Comment by TwinProduction** [comment]
@camilonova Would it be a global maintenance window, or a maintenance window per service? (not sure if that's what you meant by `a server (or many) goes down`)

It should be fairly simple to implement a global maintenance window.

**Comment by hmmcclish** [comment]
I would also love to be able to set maintenance windows, ideally per service, but globally would still help too. 

I use Gatus to alert in Discord on status changes for both my network equipment, and some services running in a Docker VM. The Docker VM is set to update at 4 AM, so I get pretty frequent messages at that time. 

Would ideally still like to know when my network goes down at 4 AM, by being able to set service-specific maintenance windows. But it would be a big help just to be able to set a global maintenance window until 4:15 AM, to avoid noise from an intentional "outage" for updates.

**Comment by TwinProduction** [comment]
Thinking about adding something like this:
```yaml
maintenance:
  timezone: America/New_York
  time: 04:00-04:15
  days: monday,wednesday     # defaults to "tuesday,wednesday,thursday,friday,saturday,sunday"
```

**Comment by TwinProduction** [comment]
This is the final format I went with after all:
```yaml
maintenance:
  start: 23:00
  duration: 1h
  every: [Monday, Thursday]
```

I was initially going to implement the `timezone` parameter, but I ended up just getting angry at the existence of timezones 😩

**Comment by TwinProduction** [comment]
Released in [v3.2.0](https://github.com/TwinProduction/gatus/releases/tag/v3.2.0)
